src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -150,71 +188,139 @@ def validate_public_ip_type(namespace): # pylint: disable=unused-argument
-            try:
-                namespace.cert_data = base64_data.decode('utf-8')
-            except UnicodeDecodeError:
-                namespace.cert_data = str(base64_data)
+    # determine if subnet is name or ID
+    is_id = is_valid_resource_id(namespace.subnet)
 
-        # change default to frontend port 443 for https
-        namespace.http_listener_protocol = 'https'
-        if not namespace.frontend_port:
-            namespace.frontend_port = 443
+    # error if vnet-name is provided along with a subnet ID
+    if is_id and namespace.virtual_network_name:
+        raise argparse.ArgumentError(None, 'Please omit --vnet-name when specifying a subnet ID')
+    elif not is_id and not namespace.virtual_network_name:
+        raise argparse.ArgumentError(None,
+                                     'Please specify --vnet-name when specifying a subnet name')
+    if not is_id:
+        namespace.subnet = resource_id(
+            subscription=get_subscription_id(),
+            resource_group=namespace.resource_group_name,
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=namespace.virtual_network_name,
+            child_type='subnets',
+            child_name=namespace.subnet)
 
 # COMMAND NAMESPACE VALIDATORS
 
+def process_ag_listener_create_namespace(namespace): # pylint: disable=unused-argument
+    if not is_valid_resource_id(namespace.frontend_ip):
+        namespace.frontend_ip = _generate_ag_subproperty_id(
+            resource_group=namespace.resource_group_name,
+            application_gateway_name=namespace.application_gateway_name,
+            child_type='frontendIpConfigurations',
+            child_name=namespace.frontend_ip)
+
+    if not is_valid_resource_id(namespace.frontend_port):
+        namespace.frontend_port = _generate_ag_subproperty_id(
+            resource_group=namespace.resource_group_name,
+            application_gateway_name=namespace.application_gateway_name,
+            child_type='frontendPorts',
+            child_name=namespace.frontend_port)
+
+    if not is_valid_resource_id(namespace.ssl_cert):
+        namespace.ssl_cert = _generate_ag_subproperty_id(
+            resource_group=namespace.resource_group_name,
+            application_gateway_name=namespace.application_gateway_name,
+            child_type='sslCertificates',
+            child_name=namespace.ssl_cert)
+
+def process_ag_http_settings_create_namespace(namespace): # pylint: disable=unused-argument
+    if not is_valid_resource_id(namespace.probe):
+        namespace.probe = _generate_ag_subproperty_id(
+            resource_group=namespace.resource_group_name,
+            application_gateway_name=namespace.application_gateway_name,
+            child_type='probes',
+            child_name=namespace.probe)
+
+def process_ag_rule_create_namespace(namespace): # pylint: disable=unused-argument
+    if not is_valid_resource_id(namespace.address_pool):
```

Comment:
You are right--it is because address-pool is mandatory here. For URL-path-map rules, address-pool is optional, because the URL-path-map itself has a default address-pool, which I use if this is not specified.

---
In reply to: [74097380](https://github.com/Azure/azure-cli/pull/616#discussion_r74097380) [](ancestors = 74097380)